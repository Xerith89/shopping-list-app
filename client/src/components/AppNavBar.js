import React, {Component} from 'react';
import  {
    Collapse, Navbar, NavbarToggler,
    Nav, NavbarBrand, NavItem, NavLink, Container  
} from 'reactstrap'

class AppNavBar extends Component {
    state = {
        isOpen: false
    }

    toggle = () => {
       this.setState({
        isOpen: !this.state.isOpen
       });
    }

    render(){
     return (  <div>
            <Navbar color="dark" dark expand="sm" className="mb-5">
            <Container>
                <NavbarBrand href="/">Posts</NavbarBrand>
                <NavbarToggler onClick={this.toggle}/>
                <Collapse isOpen={this.state.isOpen} navbar>
                    <Nav className="ml-auto" navbar>
                        <NavItem>
                            <NavLink href="#">Stuff</NavLink>
                        </NavItem>
                    </Nav>
                </Collapse>
            </Container>
            </Navbar>
        </div>);
    }
}
export {Collapse};
export default AppNavBar;