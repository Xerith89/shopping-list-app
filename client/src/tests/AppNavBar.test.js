import React from 'react';
import { render} from '@testing-library/react'
import '@testing-library/react/cleanup-after-each';
import AppNavBar from './../components/AppNavBar';

it('renders without crashing', () => {
  const element = render(<AppNavBar />)
  expect(element).toMatchSnapshot();
});
