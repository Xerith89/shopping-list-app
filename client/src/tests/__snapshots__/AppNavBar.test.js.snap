// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="mb-5 navbar navbar-expand-sm navbar-dark bg-dark"
        >
          <div
            class="container"
          >
            <a
              class="navbar-brand"
              href="/"
            >
              Posts
            </a>
            <button
              class="navbar-toggler"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse"
            >
              <ul
                class="ml-auto navbar-nav"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="#"
                  >
                    Stuff
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="mb-5 navbar navbar-expand-sm navbar-dark bg-dark"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            Posts
          </a>
          <button
            class="navbar-toggler"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
          >
            <ul
              class="ml-auto navbar-nav"
            >
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#"
                >
                  Stuff
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
